spyboat/phase_fields/gradient_descent.py

```diff
@@ -21,46 +21,92 @@
 plt.imshow(movie[frame], interpolation='nearest', cmap = 'seismic')
 
 
+def instructions(vy, vx):
+    # calculate the vector angle from and instruct the direction 
+    angle_boxes = np.array([-157.5, -112.5, -67.5, -22.5,\
+    22.5, 67.5, 112.5, 157.5, 180])
+    
+    crawl_angle = np.arctan2(vy, vx)* 180 / np.pi
+
+    # directions = np.array(['left', 'left_down', 'down', 'right_down',\
+    #  'right','right_up', 'up', 'left_up', 'left'])
+    directions = np.array([(-1,0),(-1, -1),(0, -1), (+1, -1),\
+    (+1, 0), (+1, +1), (0, +1), (-1, +1), (-1, 0)])
+
+    instruction = directions[np.where(crawl_angle<angle_boxes)][0]
```

Comment:
Here we could be more explicit by 1st getting the index of the 'directions', the last `[0]` is confusing

Reply:
done